controllers/cloud.redhat.com/providers/database/localdb.go

```diff
@@ -28,7 +28,71 @@ func NewLocalDBProvider(p *p.Provider) (providers.ClowderProvider, error) {
 // CreateDatabase ensures a database is created for the given app.  The
 // namespaced name passed in must be the actual name of the db resources
 func (db *localDbProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error {
-	if app.Spec.Database.Name == "" {
+	if app.Spec.Database.Name == "" && app.Spec.Database.SharedDBAppName == "" {
+		return nil
+	}
+
+	if app.Spec.Database.SharedDBAppName != "" {
+
+		found := false
```

Comment:
This chunk of code seems to be copypasta, but I don't know if it's enough to justify a common function.